libfreerdp/core/peer.c

```diff
@@ -818,7 +818,7 @@ BOOL freerdp_peer_context_new(freerdp_peer* client)
 		goto fail_error_description;
 	}
 
-	if (!transport_attach(rdp->transport, client->sockfd))
+	if (!rdp->update->io->TransportAttach(rdp->transport, client->sockfd))
```

Comment:
I don´t like the exposure of the internal structure of `rdpTransport` you´re doing here.
We are working on moving all module/components to be self contained with only an opaque structure exposed, this moves in the opposite direction.